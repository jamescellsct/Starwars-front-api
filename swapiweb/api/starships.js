import express from "express";

const router = express.Router();

var starships = require("./fakeData/db.json").starships;

router.get("/starships", function(req, res) {
  return res.status(200).json({
    count: starships.count,
    results: starships.results[0]
  });
});

router.get("/starships/:page", function(req, res) {
  let { page } = req.params;
  page = page.slice(5, 6) - 1;

  if (!starships.results[page]) {
    return res.status(404).send({
      error: {
        message: "Page Not Found!"
      }
    });
  }

  return res.status(200).json({
    count: starships.count,
    results: starships.results[page]
  });
});

export default router;
