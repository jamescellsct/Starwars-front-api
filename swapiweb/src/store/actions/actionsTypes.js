//Peoples Actions Types
const FETCH_PEOPLE_STARTED = "FETCH_PEOPLE_STARTED";
const FETCH_PEOPLE_SUCCEEDED = "FETCH_PEOPLE_SUCCEEDED";
const FETCH_PEOPLE_FAILED = "FETCH_PEOPLE_FAILED";
const UPDATE_PEOPLE_IS_FETCHING = "UPDATE_PEOPLE_IS_FETCHING";
const UPDATE_PEOPLE_SUCCEEDED = "UPDATE_PEOPLE_SUCCEEDED";
const UPDATE_PEOPLE_FAILED = "UPDATE_PEOPLE_FAILED";
export const peoplesAction = {
  FETCH_PEOPLE_STARTED,
  FETCH_PEOPLE_SUCCEEDED,
  FETCH_PEOPLE_FAILED,
  UPDATE_PEOPLE_IS_FETCHING,
  UPDATE_PEOPLE_SUCCEEDED,
  UPDATE_PEOPLE_FAILED
};

//Planets Actions Types
const FETCH_PLANETS_STARTED = "FETCH_PLANETS_STARTED";
const FETCH_PLANETS_SUCCEEDED = "FETCH_PLANETS_SUCCEEDED";
const FETCH_PLANETS_FAILED = "FETCH_PLANETS_FAILED";
const UPDATE_PLANETS_IS_FETCHING = "UPDATE_PLANETS_IS_FETCHING";
const UPDATE_PLANETS_SUCCEEDED = "UPDATE_PLANETS_SUCCEEDED";
const UPDATE_PLANETS_FAILED = "UPDATE_PLANETS_FAILED";
export const planetsAction = {
  FETCH_PLANETS_STARTED,
  FETCH_PLANETS_SUCCEEDED,
  FETCH_PLANETS_FAILED,
  UPDATE_PLANETS_IS_FETCHING,
  UPDATE_PLANETS_SUCCEEDED,
  UPDATE_PLANETS_FAILED
};

//Starships Actions Types
const FETCH_STARSHIPS_STARTED = "FETCH_STARSHIPS_STARTED";
const FETCH_STARSHIPS_SUCCEEDED = "FETCH_STARSHIPS_SUCCEEDED";
const FETCH_STARSHIPS_FAILED = "FETCH_STARSHIPS_FAILED";
const UPDATE_STARSHIPS_IS_FETCHING = "UPDATE_STARSHIPS_IS_FETCHING";
const UPDATE_STARSHIPS_SUCCEEDED = "UPDATE_STARSHIPS_SUCCEEDED";
const UPDATE_STARSHIPS_FAILED = "UPDATE_STARSHIPS_FAILED";
export const starshipsAction = {
  FETCH_STARSHIPS_STARTED,
  FETCH_STARSHIPS_SUCCEEDED,
  FETCH_STARSHIPS_FAILED,
  UPDATE_STARSHIPS_IS_FETCHING,
  UPDATE_STARSHIPS_SUCCEEDED,
  UPDATE_STARSHIPS_FAILED
};

//Vehicles Actions Types
const FETCH_VEHICLES_STARTED = "FETCH_VEHICLES_STARTED";
const FETCH_VEHICLES_SUCCEEDED = "FETCH_VEHICLES_SUCCEEDED";
const FETCH_VEHICLES_FAILED = "FETCH_VEHICLES_FAILED";
const UPDATE_VEHICLES_IS_FETCHING = "UPDATE_VEHICLES_IS_FETCHING";
const UPDATE_VEHICLES_SUCCEEDED = "UPDATE_VEHICLES_SUCCEEDED";
const UPDATE_VEHICLES_FAILED = "UPDATE_VEHICLES_FAILED";
export const vehiclesAction = {
  FETCH_VEHICLES_STARTED,
  FETCH_VEHICLES_SUCCEEDED,
  FETCH_VEHICLES_FAILED,
  UPDATE_VEHICLES_IS_FETCHING,
  UPDATE_VEHICLES_SUCCEEDED,
  UPDATE_VEHICLES_FAILED
};

//Pagination Action Types
const PAGE_SELECTED = "PAGE_SELECTED";
export const paginationAction = {
  PAGE_SELECTED
};
