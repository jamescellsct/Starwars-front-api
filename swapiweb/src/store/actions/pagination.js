import { paginationAction } from "./actionsTypes";

const pageSelected = pagination => ({
  type: paginationAction.PAGE_SELECTED,
  pagination
});

export const selectPagePagination = page => dispatch => {
  dispatch(
    pageSelected({
      page
    })
  );
};
