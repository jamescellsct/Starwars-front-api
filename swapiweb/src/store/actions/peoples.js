import { peoplesAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchPeopleStarted = () => ({
  type: peoplesAction.FETCH_PEOPLE_STARTED
});

const fetchPeopleSucceeded = people => ({
  type: peoplesAction.FETCH_PEOPLE_SUCCEEDED,
  people
});

const fetchPeopleFailed = error => ({
  type: peoplesAction.FETCH_PEOPLE_FAILED,
  error
});

export const fetchPeople = () => dispatch => {
  dispatch(fetchPeopleStarted());

  swapiRequest(null, null)
    .then(response => {
      dispatch(
        fetchPeopleSucceeded({
          results: response.results,
          total: response.count
        })
      );
    })
    .catch(error => {
      dispatch(fetchPeopleFailed(error));
    });
};
