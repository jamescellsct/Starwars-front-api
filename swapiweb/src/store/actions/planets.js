import { planetsAction } from "./actionsTypes";
import { getPlanets } from "../../services/api/planetsService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchPlanetsStarted = () => ({
  type: planetsAction.FETCH_PLANETS_STARTED
});

const fetchPlanetsSucceeded = planets => ({
  type: planetsAction.FETCH_PLANETS_SUCCEEDED,
  planets
});

const fetchPlanetsFailed = error => ({
  type: planetsAction.FETCH_PLANETS_FAILED,
  error
});

export const fetchPlanets = () => dispatch => {
  dispatch(fetchPlanetsStarted());

  getPlanets()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchPlanetsSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchPlanetsFailed(error));
    });
};

const updatePlanetsIsFetching = () => ({
  type: planetsAction.UPDATE_PLANETS_IS_FETCHING
});

const updatePlanetsSucceeded = planets => ({
  type: planetsAction.UPDATE_PLANETS_SUCCEEDED,
  planets
});

const updatePlanetsFailed = error => ({
  type: planetsAction.UPDATE_PLANETS_FAILED,
  error
});

export const updatePlanets = page => dispatch => {
  dispatch(updatePlanetsIsFetching());
  getPlanets(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updatePlanetsSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updatePlanetsFailed(error));
    });
};
