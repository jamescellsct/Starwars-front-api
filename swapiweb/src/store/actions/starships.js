import { starshipsAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchStarshipsStarted = () => ({
  type: starshipsAction.FETCH_STARSHIPS_STARTED
});

const fetchStarshipsSucceeded = starships => ({
  type: starshipsAction.FETCH_STARSHIPS_SUCCEEDED,
  starships
});

const fetchStarshipsFailed = error => ({
  type: starshipsAction.FETCH_STARSHIPS_FAILED,
  error
});

export const fetchStarships = () => dispatch => {
  dispatch(fetchStarshipsStarted());

  swapiRequest(null, null, "starships")
    .then(response => {
      dispatch(
        fetchStarshipsSucceeded({
          results: response.results,
          total: response.count
        })
      );
    })
    .catch(error => {
      dispatch(fetchStarshipsFailed(error));
    });
};
