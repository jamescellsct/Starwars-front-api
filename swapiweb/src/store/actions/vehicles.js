import { vehiclesAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchVehiclesStarted = () => ({
  type: vehiclesAction.FETCH_VEHICLES_STARTED
});

const fetchVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.FETCH_VEHICLES_SUCCEEDED,
  vehicles
});

const fetchVehiclesFailed = error => ({
  type: vehiclesAction.FETCH_VEHICLES_FAILED,
  error
});

export const fetchVehicles = () => dispatch => {
  dispatch(fetchVehiclesStarted());

  swapiRequest(null, null, "vehicles")
    .then(response => {
      dispatch(
        fetchVehiclesSucceeded({
          results: response.results,
          page: response.page,
          pageCount: response.pageCount
        })
      );
    })
    .catch(error => {
      dispatch(fetchVehiclesFailed(error));
    });
};

const updateVehiclesIsFetching = () => ({
  type: vehiclesAction.UPDATE_VEHICLES_IS_FETCHING
});

const updateVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.UPDATE_VEHICLES_SUCCEEDED,
  vehicles
});

const updateVehiclesFailed = error => ({
  type: vehiclesAction.UPDATE_VEHICLES_FAILED,
  error
});

export const updateVehicles = page => dispatch => {
  dispatch(updateVehiclesIsFetching());
  swapiRequest(null, page, "vehicles")
    .then(response => {
      dispatch(
        updateVehiclesSucceeded({
          results: response.results,
          page: response.page,
          pageCount: response.pageCount
        })
      );
    })
    .catch(error => {
      dispatch(updateVehiclesFailed(error));
    });
};
