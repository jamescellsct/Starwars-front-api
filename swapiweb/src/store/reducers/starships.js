import { starshipsAction } from "../actions/actionsTypes";

export default function starships(state = null, action) {
  switch (action.type) {
    case starshipsAction.FETCH_STARSHIPS_STARTED:
      return Object.assign({}, state, {
        isFetching: true
      });

    case starshipsAction.FETCH_STARSHIPS_SUCCEEDED:
      return Object.assign({}, state, {
        list: [...(state.list || []), ...action.starships.results],
        isFetching: false
      });

    case starshipsAction.FETCH_STARSHIPS_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    default:
      return state;
  }
}
