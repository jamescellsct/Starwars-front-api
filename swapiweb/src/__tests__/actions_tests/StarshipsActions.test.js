import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
import nock from "nock";

import { starshipsAction } from "../../store/actions/actionsTypes";
import { fetchStarships } from "../../store/actions/starships";
import BASE_URL from "./baseUrl";

const midlewares = [thunk];
const mockStore = configureMockStore(midlewares);

const mockResults = [
  {
    name: "Dale Jackson",
    avatar: "http://i.pravatar.cc/40?img=0.7474128780010723",
    model: "so",
    manufacturer: "vajcez",
    length: 191,
    max_atmosphering_speed: 205,
    passengers: 1473,
    cargo_capacity: 5579622,
    consumables: 1,
    starship_class: "rukvu",
    idItem: 1
  },
  {
    name: "Sylvia Barton",
    avatar: "http://i.pravatar.cc/40?img=0.7575125938238629",
    model: "ejvoj",
    manufacturer: "nubodivas",
    length: 17,
    max_atmosphering_speed: 522,
    passengers: 96,
    cargo_capacity: 2861477,
    consumables: 5,
    starship_class: "lu",
    idItem: 2
  }
];

describe("Testing Starships Action", () => {
  it("Shold create an action to fetching starships data", () => {
    afterEach(() => {
      nock.cleanAll();
    });

    nock(BASE_URL)
      .defaultReplyHeaders({ "access-control-allow-origin": "*" })
      .get("/starships")
      .reply(200, {
        count: 20,
        results: mockResults
      });

    const expectedAction = [
      { type: starshipsAction.FETCH_STARSHIPS_STARTED },
      {
        type: starshipsAction.FETCH_STARSHIPS_SUCCEEDED,
        starships: {
          page: 1,
          results: mockResults,
          pageCount: 2
        }
      }
    ];

    const store = mockStore({ starships: {} });
    return store.dispatch(fetchStarships()).then(() => {
      expect(store.getActions()).toEqual(expectedAction);
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
