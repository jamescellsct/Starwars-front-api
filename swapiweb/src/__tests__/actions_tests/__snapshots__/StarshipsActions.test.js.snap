// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Starships Action Shold create an action to fetching starships data 1`] = `
Array [
  Object {
    "type": "FETCH_STARSHIPS_STARTED",
  },
  Object {
    "starships": Object {
      "page": 1,
      "pageCount": 2,
      "results": Array [
        Object {
          "avatar": "http://i.pravatar.cc/40?img=0.7474128780010723",
          "cargo_capacity": 5579622,
          "consumables": 1,
          "idItem": 1,
          "length": 191,
          "manufacturer": "vajcez",
          "max_atmosphering_speed": 205,
          "model": "so",
          "name": "Dale Jackson",
          "passengers": 1473,
          "starship_class": "rukvu",
        },
        Object {
          "avatar": "http://i.pravatar.cc/40?img=0.7575125938238629",
          "cargo_capacity": 2861477,
          "consumables": 5,
          "idItem": 2,
          "length": 17,
          "manufacturer": "nubodivas",
          "max_atmosphering_speed": 522,
          "model": "ejvoj",
          "name": "Sylvia Barton",
          "passengers": 96,
          "starship_class": "lu",
        },
      ],
    },
    "type": "FETCH_STARSHIPS_SUCCEEDED",
  },
]
`;
